src/pages/test-report/cat-d/components/reverse-diagram-modal/__tests__/reverse-diagram-modal.cat-d.spec.ts

```diff
@@ -0,0 +1,169 @@
+        StoreModule.forRoot({
+          tests: () => ({
+            currentTest: {
+              slotId: '123',
+            },
+            testStatus: {},
+            startedTests: {
+              123: {
+                category: TestCategory.D,
+                vehicleDetails: {
+                  vehicleLength: 10,
+                  vehicleWidth: 2.75,
+                },
+                accompaniment: {},
+                testData: {
+                  dangerousFaults: {},
+                  drivingFaults: {},
+                  manoeuvres: {},
+                  seriousFaults: {},
+                  testRequirements: {},
+                  ETA: {},
+                  eco: {},
+                  vehicleChecks: {
+                    showMeQuestions: [{
+                      code: 'S3',
+                      description: '',
+                      outcome: '',
+                    }],
+                    tellMeQuestions: [{
+                      code: '',
+                      description: '',
+                      outcome: '',
+                    }],
+                  },
+                  eyesightTest: {},
+                },
+                activityCode: '28',
+                journalData: {
+                  candidate: {
+                    candidateName: 'Joe Bloggs',
+                    driverNumber: '123',
+                  },
+                },
+                rekey: false,
+              },
+            },
+          }),
+        }),
+      ],
+      providers: [
+        { provide: Config, useFactory: () => ConfigMock.instance() },
+        { provide: Platform, useFactory: () => PlatformMock.instance() },
+        { provide: NavParams, useFactory: () => NavParamsMock.instance() },
+        { provide: App, useClass: MockAppComponent },
+        ReversingDistancesProvider,
+      ],
+    });
+  });
+
+  beforeEach(async(() => {
+    fixture = TestBed.createComponent(ReverseDiagramCatDPage);
+    component = fixture.componentInstance;
+    store$ = TestBed.get(Store);
+    store$.dispatch(new StartTest(123, TestCategory.D));
+  }));
+
+  describe('DOM', () => {
+
+  });
+
+  describe('Class', () => {
+
+    const vehicleDetails: CatCUniqueTypes.VehicleDetails = {
+      vehicleLength: 10,
+      vehicleWidth: 2.75,
+    };
+
+    describe('ngOnInit', () => {
+      it('should set the distance based on booked in vehicle length', (done: DoneFn) => {
+        component.ngOnInit();
+        component.componentState.vehicleLength$.subscribe((result) => {
+          expect(result).toEqual(10);
+          done();
+        });
+      });
+
+      it('should set the distance based on booked in vehicle width', (done: DoneFn) => {
+        component.ngOnInit();
+        component.componentState.vehicleWidth$.subscribe((result) => {
+          expect(result).toEqual(2.75);
+          done();
+        });
+      });
+    });
+
+    // TODO Check that values are correct for Cat D (Refactored to make tests pass!)
```

Comment:
I guess the default values are being returned because there is no match on the test category, hence the refactor.
We could update the reversing distances provider as part of this PR which would negate the need for the refactor. I appreciate you may have wanted to update the provider as part of https://jira.dvsacloud.uk/browse/MES-4122

Reply:
Will do this one as part of 4122 as I am nearing completion of this ticket; but i'll move straight onto that after this. Just waiting for develop to be fixed so I can manually test & then push the changes required for this one. All the other changes have been dealt with already & theres a few missing components waiting to be pushed (uncouple/recouple & vehicle checks), had forgotten to put WIP in the title and set this as a draft PR (sorry!).